lib/mongo/retryable.rb

```diff
@@ -46,18 +46,19 @@ module Retryable
     # @return [ Result ] The result of the operation.
     #
     # @since 2.1.0
-    def read_with_retry(attempt = 0, &block)
+    def read_with_retry(attempt = 0, max_read_retries = nil, &block)
+      max_retries = max_read_retries || cluster.max_read_retries
       begin
         block.call
       rescue Error::SocketError, Error::SocketTimeoutError
         retry_operation(&block)
       rescue Error::OperationFailure => e
         if cluster.sharded? && e.retryable?
```

Comment:
After trying to make the Retryable module more generic and avoid references to cluster, I realized it made the API and design pretty messy; in every class that included the Retryable module, or inherited it, I had to forward the methods like #max_read_retries and #read_retry_interval to cluster. I finally realized that it would be much more simple and explicit to make a new method, Retryable#read_with_one_retry and use that when calling ismaster on a Monitor's socket.
The PR is here: https://github.com/mongodb/mongo-ruby-driver/pull/769